core/src/main/scala/pureconfig/DerivedReaders.scala

```diff
@@ -20,9 +20,26 @@ import shapeless.labelled._
 trait AllowMissingKey { self: ConfigReader[_] => }
 
 /**
+ * Trait extending [[DerivedReaders1]] that contains `ConfigReader` instances for `AnyVal`.
+ *
+ * This trait exists to give priority to the `AnyVal` derivation over the generic product derivation.
+ */
+trait DerivedReaders extends DerivedReaders1 {
```

Comment:
Is the `<:!<` constraint provided by some library we require in the core module, or a trick we would implement? If it's the latter, then I prefer the two traits solution.